packages/express-backend/routes/entries.js

```diff
@@ -1,68 +1,130 @@
-//entries.js 
+//entries.js
 
 import express from "express";
-const router = express.Router()
+const router = express.Router();
 import { createClient } from "@supabase/supabase-js";
 import dotenv from "dotenv";
+import { authenticateUser } from "../auth.js";
 dotenv.config({ path: "../../.env" });
 
 const supabaseUrl = "https://vzutkihkzjyhnwzqsgrx.supabase.co";
 const supabaseKey = process.env.SUPABASE_KEY;
 const supabase = createClient(supabaseUrl, supabaseKey);
 
-//returns entry given an id
-router.get("/:id", async (req, res) => {
-    try {
-      const { data, error } = await supabase
-        .from("entries")
-        .select("*")
-        .filter("entry_id", "eq", req.params["id"]);
-      if (error) {
-        return res.status(500).send({ error: error.message });
-      }
-      res.status(200).send(data);
-    } catch (err) {
-      res.status(500).send({ error: "Server error" });
+router.get("/home", authenticateUser, async (req, res) => {
+  console.log(req.user_name);
+  try {
+    const { data, error } = await supabase
+      .from("main_page")
+      .select("*")
+      .eq("user_name", req.user_name);
+    if (error) {
+      res.status(400).send({ error: error.message });
     }
-  });
-  
-  //used to look up all public entries
-  router.get("/", async (req, res) => {
-    const is_public = req.query.is_public ? req.query.is_public : "False";
-    try {
-      const { data, error } = await supabase.from("public_entries").select("*");
-      if (error) {
-        return res.status(500).send({ error: error.message });
-      }
-      console.log(data);
-      res.status(200).send(data);
+    res.status(200).send(data);
     } catch (err) {
-      res.status(500).send({ error: "Server error" });
+    res.status.send({ error: "Server error " });
+  }
+});
+
+router.get("/home", authenticateUser, async (req, res) => {
+  console.log(req.user_name);
+  try {
+    const { data, error } = await supabase
+      .from("main_page")
```

Comment:
Do we need to use another view for this? why can't we use the existing entries table? All we'd have to do is update our select in the query.